params_test.go

```diff
@@ -20,3 +22,66 @@ func TestFormatTime(test *testing.T) {
 	formattedTime := formatTime(time)
 	t.Assert(formattedTime, formatTime(time), "formatTime()")
 }
+
+func TestParamsWithOptions(test *testing.T) {
+	t := &T{test}
+
+	ctx := context.Background()
+	iKey := "asdf"
+	p := &Params{}
+	p.ApplyOptions([]Option{
+		WithIdempotencyKey(iKey),
+		WithContext(ctx),
+	})
+
+	t.Assert(iKey, p.IdempotencyKey, "ApplyOptions()")
+	t.Assert(ctx, p.Context, "ApplyOptions()")
+}
+
+func TestCommonParamsWithOptions(test *testing.T) {
```

Comment:
let's add a test case where no options are passed in, to make sure we dont hit any nil pointer issues. here's how i like to set up multiple test cases within one test function. by calling t.Run() it will treat each iteration as a unique test, and it saves you on LOC's https://github.com/recurly/callbacks/blob/c6edd0ce8e5d16bb759989794c60f60592a94214/api/callbacks/mime_test.go#L11